garage/tf/policies/categorical_gru_policy.py

```diff
@@ -86,19 +86,24 @@ def __init__(
             if feature_network is None:
                 feature_var = flat_input_var
             else:
-                feature_var = L.get_output(
-                    l_flat_feature,
-                    {feature_network.input_layer: flat_input_var})
-
-            self.f_step_prob = tensor_utils.compile_function(
-                [
-                    flat_input_var,
-                    prob_network.step_prev_hidden_layer.input_var
-                ],
-                L.get_output([
-                    prob_network.step_output_layer,
-                    prob_network.step_hidden_layer
-                ], {prob_network.step_input_layer: feature_var}))
+                with tf.name_scope("feature_network", values=[flat_input_var]):
+                    feature_var = L.get_output(
+                        l_flat_feature,
+                        {feature_network.input_layer: flat_input_var})
+
+            with tf.name_scope(self._prob_network_name, values=[feature_var]):
```

Comment:
In this case I'm reusing the same name to produce different instances of the same network. Each set of network operations under name scope is different since their inputs and output are connected to different places in the graph, but they perform the same symbolic operations as the network they're instantiated from.